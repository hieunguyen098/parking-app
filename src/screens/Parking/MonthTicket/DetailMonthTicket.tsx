import { StyleSheet, Text, View, Image, Modal, TouchableOpacity, Button, TextInput } from 'react-native';
import React, { useState } from 'react';
import BottomButton from '../../../components/Buttons/BottomButton';
import FieldValue from '../../../components/FieldValue';
import { GlobalStyles } from '../../../constants';
import SmallButton from '../../../components/Buttons/SmallButton';
import ModalExtendMonthTicket from './ModalExtendMonthTicket';
import ModalGiveTicket from './ModalGiveTicket';
import moment from "moment";

const DetailMonthTicket = ({route} : any) => {
    const { item } = route.params;
    const [extendNumber, setExtendNumber] = useState(0);
    const [expandTicket, setExpandTicket] = useState(false);
    const [giveTicket, setGiveTicket] = useState(false);

    const handleGiveTicket = () => {
        setGiveTicket(true);
    };

    const handleExpandTicket = () => {
        setExpandTicket(true);
        console.log('Expand ticket');
    };

    const convertDueDate = (numOfMonth: string, startDate: string) => {
        if (!isNaN(parseInt(numOfMonth))) {
            return moment(startDate).add(parseInt(numOfMonth), "month").format("DD/MM/YYYY");
        } else {
            return "NaN"
        }
    }

    return (
        <>
            <ModalExtendMonthTicket item = {item} expandTicket={expandTicket} setExpandTicket={setExpandTicket} extendNumber={extendNumber} setExtendNumber={setExtendNumber}/>
            <ModalGiveTicket item={item} giveTicket={giveTicket} setGiveTicket={setGiveTicket} />
            <View style={styles.container}>
                <View style={styles.boxContent}>
                    <View style={styles.top}>
                        <View style={styles.imageContainer}>
                            <Image
                                source={{
                                    uri: item.locationImageUrl ? item.locationImageUrl : "",
                                }}
                                style={styles.image}
                            />
                        </View>
                        <View style={styles.content}>
                            <Text style={styles.title}>{item.locationName}</Text>
                            <SmallButton
                                title={'Tặng'}
                                style={{ width: 80, marginLeft: 'auto' }}
                                onPress={handleGiveTicket}
                            />
                        </View>
                    </View>
                    <FieldValue fieldName="Số tháng đã mua" value={parseInt(item.number) + extendNumber} />
                    <FieldValue fieldName="Địa chỉ" value={item.locationAddress} />
                    <FieldValue fieldName="Ngày mua" value={moment(item.startDate).format("DD/MM/YYYY")} />
                    <FieldValue fieldName="Có giá trị đến hết ngày" value={convertDueDate(item.number, item.startDate)} />
                </View>
            </View>
            <BottomButton title="Gia hạn" onPress={handleExpandTicket} />
        </>
    );
};

export default DetailMonthTicket;

const styles = StyleSheet.create({
    container: {
        flex: 1,
        padding: 20,
    },
    boxContent: {
        backgroundColor: '#fff',
        borderRadius: 20,
        padding: 20,
    },
    top: {
        flexDirection: 'row',
        borderBottomWidth: 1,
        borderColor: GlobalStyles.colors.lightGrey100,
        paddingBottom: 10,
    },
    imageContainer: {
        marginRight: 8,
    },
    image: {
        width: 70,
        height: 70,
        borderRadius: 8,
    },
    content: {
        flex: 1,
        justifyContent: 'space-between',
    },
    title: {
        fontSize: 16,
        fontWeight: '500',
    },
});
