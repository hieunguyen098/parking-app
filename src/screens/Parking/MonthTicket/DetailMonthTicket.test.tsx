import * as React from 'react';
import { render } from '@testing-library/react-native';

import DetailMonthTicket from './DetailMonthTicket';

it('renders correctly', () => {
    const tree = render(<DetailMonthTicket />);
    expect(tree).toMatchSnapshot();
});
