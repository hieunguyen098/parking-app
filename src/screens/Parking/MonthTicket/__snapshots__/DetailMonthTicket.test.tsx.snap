// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  />,
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  />,
  <View
    style={
      Object {
        "flex": 1,
        "padding": 20,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#fff",
          "borderRadius": 20,
          "padding": 20,
        }
      }
    >
      <View
        style={
          Object {
            "borderBottomWidth": 1,
            "borderColor": "#E8E8E8",
            "flexDirection": "row",
            "paddingBottom": 10,
          }
        }
      >
        <View
          style={
            Object {
              "marginRight": 8,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://upload.wikimedia.org/wikipedia/commons/1/19/Blue_Disc_Parking_Area_Markings_Blue_Paint.JPG",
              }
            }
            style={
              Object {
                "borderRadius": 8,
                "height": 70,
                "width": 70,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "500",
              }
            }
          >
            Nhà xe trường đại học Bách Khoa TP. HCM
          </Text>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FF953A",
                  "borderRadius": 8,
                  "justifyContent": "center",
                  "minWidth": 40,
                  "paddingHorizontal": 12,
                  "paddingVertical": 6,
                },
                Object {
                  "marginLeft": "auto",
                  "width": 80,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 16,
                    "fontWeight": "600",
                    "textAlign": "center",
                  },
                  undefined,
                ]
              }
            >
              Tặng
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginVertical": 10,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "gray",
              "fontSize": 16,
              "fontWeight": "500",
              "width": "30%",
            }
          }
        >
          Địa chỉ
        </Text>
        <Text
          style={
            Object {
              "fontSize": 16,
              "textAlign": "right",
              "width": "70%",
            }
          }
        >
          Dĩ An, Bình Dương
        </Text>
      </View>
      <View
        style={
          Object {
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginVertical": 10,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "gray",
              "fontSize": 16,
              "fontWeight": "500",
              "width": "30%",
            }
          }
        >
          Ngày mua
        </Text>
        <Text
          style={
            Object {
              "fontSize": 16,
              "textAlign": "right",
              "width": "70%",
            }
          }
        >
          01/09/2022
        </Text>
      </View>
      <View
        style={
          Object {
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginVertical": 10,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "gray",
              "fontSize": 16,
              "fontWeight": "500",
              "width": "30%",
            }
          }
        >
          Có giá trị đến hết ngày
        </Text>
        <Text
          style={
            Object {
              "fontSize": 16,
              "textAlign": "right",
              "width": "70%",
            }
          }
        >
          30/11/2022
        </Text>
      </View>
    </View>
  </View>,
  <View
    style={
      Object {
        "borderTopLeftRadius": 24,
        "borderTopRightRadius": 24,
        "bottom": 0,
        "elevation": 1,
        "padding": 20,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "#FF953A",
            "borderRadius": 8,
            "paddingHorizontal": 12,
            "paddingVertical": 18,
            "width": "100%",
          },
          Object {
            "backgroundColor": "#FF953A",
          },
          undefined,
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 18,
              "fontWeight": "600",
              "textAlign": "center",
            },
            Object {
              "color": "#fff",
            },
            undefined,
          ]
        }
      >
        Gia hạn
      </Text>
    </View>
  </View>,
]
`;
